Fix pybind11 warnings in python_rpc_handler.cpp

pietern opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27531 Run clang-format for torch/distributed/rpc
* #27530 Rename PythonUDF{Call,Resp}
* #27529 Scope pybind11 functions to torch.distributed.{autograd,rpc}
* #27290 Move RPC API to torch.distributed.rpc
* #27289 Move internal functions to torch.distributed.rpc
* #27288 Move RPC backend registry to torch.distributed.rpc
* #27287 Remove torch.distributed.rpc function
* #27286 Rename variables and add comments
* #27285 Remove shebang from non-executable files in torch.distributed
* **#27284 Fix pybind11 warnings in python_rpc_handler.cpp**

The warnings related to usage of the deprecated != operator. Instead
of checking the member field on every function call, we can check it
once, on construction of PythonRpcHandler.

Differential Revision: [D17808213](https://our.internmc.facebook.com/intern/diff/D17808213)

pietern committed:
Fix pybind11 warnings in python_rpc_handler.cpp

The warnings related to usage of the deprecated != operator. Instead
of checking the member field on every function call, we can check it
once, on construction of PythonRpcHandler.

zhaojuanmao commented:
thanks for fixing it!

mrshenli reviewed (approved):
#25499 adds one more serialize function to this file. Shall also replace it with the `getFunction` wrapper?

pietern commented:
@mrshenli I'll rebase past #25499 and include the function.

pietern committed:
Update on "Fix pybind11 warnings in python_rpc_handler.cpp"


The warnings related to usage of the deprecated != operator. Instead
of checking the member field on every function call, we can check it
once, on construction of PythonRpcHandler.
